discovery-infra/node_controlers/qe_vm_controler.py

```diff
@@ -0,0 +1,94 @@
+import os
+import re
+import libvirt
+import utils
+
+from  node_controlers.node_controler import NodeControler
+
+class QeVmController(NodeControler):
+
+    def __init__(self):
+        self.libvirt_connection = libvirt.open('qemu:///system')
+
+    def __del__(self):
+        self.libvirt_connection.close()
+
+    def list_nodes(self):
+
+        nodes = {}
+        domains = self.libvirt_connection.listAllDomains()
+
+        for domain in domains:
+            domain_name = domain.name()
+            if ("master" in domain_name) or ("worker" in domain_name):
+                domain_state = "running" if domain.isActive() else "shut_off"
+                nodes[domain_name] = domain_state
+
+        return nodes
+    
+    def shutdown_node(self, node_name):
+
```

Comment:
not sure why we need spaces in those functions? Not a big deal but interesting why?

Reply:
removed